ensime-sbt.el

```diff
@@ -144,14 +151,13 @@ again."
                                       (?q "test-[q]uick" "testQuick"))))
     (concat module "/" source-set task)))
 
-(defun ensime-sbt-find-subproject (file-name source-set)
-  (when source-set
-    (let* ((config (-> (ensime-connection) ensime-config))
-           (subprojects (plist-get config :subprojects))
-           (matches-subproject-dir? (lambda (dir) (string-match-p dir file-name)))
-           (find-subproject (lambda (sp)
-                              (-any matches-subproject-dir? (plist-get sp :source-roots)))))
-      (-> (-find find-subproject subprojects) (plist-get :name)))))
+(defun ensime-sbt-find-subproject (file-name)
```

Comment:
I was reading the emacs lisp book today on the train and I discovered `locate-file`, which might be a much better way to implement this. Also, I think this function is general enough to be used outside of the ensime sbt support, perhaps rename and move to `ensime-config`?

FYI if you want to know where `->` comes from (because the `C-h f` description doesn't do it justice) it's from clojure. https://clojuredocs.org/clojure.core/-%3E (it took me to read a book on clojure to understand what was going on).